### DISCUSSION TITLE
Error during Data Export from PostgreSQL db

### DISCUSSION THREAD

**Discussion by mielune** [Budibase/budibase#8434] [Help]
Hi folks,

I tried to use the "Export Data" action in a button of my app and it raised an "Unexpected token ' in JSON at position 1" error.
Connection to the database is ok, table is working fine.

No more details in the server-side logs.
BB version 2.0.32

![image](https://user-images.githubusercontent.com/8350742/198545062-f8ca776c-d8a6-4e12-b064-e2face77633c.png)

Thank's for your help and all the work done on BB :)

**Comment by mielune** [comment]
Just tried with 2.0.37, same error.

**Comment by mielune** [comment]
More details, selected only one row; request from the http request from browser to bb backend:
![image](https://user-images.githubusercontent.com/8350742/198688503-dc69f0cc-6c3e-4a9d-8e6c-a1d87635011b.png)

Payload:
![image](https://user-images.githubusercontent.com/8350742/198688769-ef574b06-6dd2-4707-9e83-b0c585de9698.png)

Response:
![image](https://user-images.githubusercontent.com/8350742/198688870-41684692-5962-4283-aca5-db8c7ac5ee9e.png)

I hope that can help.

**Comment by mielune** [comment]
Hi folks,

No-one has this trouble ?

Testing this feature in a v1.2.20 it is working fine:
![image](https://user-images.githubusercontent.com/8350742/200015546-ad6156ff-e073-46c1-92f9-d5a18fb08f10.png)
The call from the frontend to the backend contains:
![image](https://user-images.githubusercontent.com/8350742/200015184-ca6788d1-6fa5-459c-b3f2-3c0ae74e22d4.png)

In the v2.1.11 it failed:
![image](https://user-images.githubusercontent.com/8350742/200037802-cb568ea6-522b-4ea0-abdb-1b631fa37836.png)
![image](https://user-images.githubusercontent.com/8350742/200037876-91f28e4f-94e1-408c-b6d8-737f74a9758d.png)

How can I help more ?

Kind regards

**Reply by ecxs** [reply]
@mielune I have exactly the same problem. Also with a PostgreSQL (/Supabase) datasource.

**Comment by mielune** [comment]
Hi folks,

If the table primary key is an integer, it is working; but it failed with other types of data (uuid, text, etc.)

**Comment by NicolasFabre81** [comment]
For me it fails even with an integer PK : it is adding quotes where there is none supposed to be :
<img width="556" alt="image" src="https://user-images.githubusercontent.com/28592069/217584122-2ddf325c-16dc-4cb0-9bdf-41ebb915a198.png">

**Reply by NicolasFabre81** [reply]
@melohagan if you can have a look please

**Reply by melohagan** [reply]
Hey @NicolasFabre81 Could you try upgrading to **v2.3.6** please?

**Reply by NicolasFabre81** [reply]
@melohagan I just updated my app to V2.3.6, still getting the same error :
<img width="753" alt="image" src="https://user-images.githubusercontent.com/28592069/222692662-f02f6b55-eae6-42ad-bbd3-ca0ef5fcbf9e.png">

**Comment by NicolasFabre81** [comment]
@mielune @melohagan Here is an awesome workaround to get it working : https://github.com/Budibase/budibase/discussions/9032
Special thanks to @AlejandroFNadal

**Reply by AlejandroFNadal** [reply]
Thanks for mentioning it <3, I hope this can get somehow implemented into an actual feature

**Reply by NicolasFabre81** [reply]
Hello @AlejandroFNadal , could'nt we update this line to be able to catch data from a data provider, just after having it refreshed ?
It could avoid us to build a select query just to make the export.
`rawResults = $("Action 1.Query result.data")`
PS : you're not on the bb discord channel ?

**Reply by AlejandroFNadal** [reply]
> Hello @AlejandroFNadal , couldn't we update this line to be able to catch data from a data provider, just after having it refreshed ? It could avoid us to build a select query just to make the export. `rawResults = $("Action 1.Query result.data")` PS : you're not on the bb discord channel ?

Yes, that is actually a great idea! If you already have the data on a DP, just obtain it from there.
If you have the time, write it down in the workaround post as a comment so that others can find the idea as well.

Also, if you find out why it doesn't work in Firefox, I would be extra grateful :D 

I am in the discord channel, but I surprisingly feel that here (in github) there is more communication than in the ds.

**Reply by NicolasFabre81** [reply]
> just obtain it from there

My question was : how ? 8-)
I don't find enough doc about the use of JS in bb...

**Reply by NicolasFabre81** [reply]
If I try rawResults = $("myDataProvider.Rows"); it only returns the first page of rows. (My DP is paginated)
